lib/fastlane/plugin/wpmreleasetoolkit/helper/github_helper.rb

```diff
@@ -19,9 +19,13 @@ def self.github_token!
         token || UI.user_error!('Please provide a GitHub authentication token via the `GITHUB_TOKEN` environment variable')
       end
 
-      def self.github_client
+      # Creates a client for the GitHub API
+      #
+      # @param [String] accesstoken GitHub OAuth access token
+      #
+      def self.github_client(accesstoken)
         @@client ||= begin
-          client = Octokit::Client.new(access_token: github_token!)
+          client = Octokit::Client.new(access_token: accesstoken)
```

Comment:
Yeah, I did not know that error at the instance retainer, thanks for pointing that out. I agree that will makes more sense to follow your suggestion and make the `@client` instantiated at the initialization of the `GithubHelper`. To use the methods as instance methods.

Also, just to confirm if I understand everything correctly. At the initialization, I'll use the same github_client that is there today, without the memoization. So at the init, I will have this:
```
def initialize(access_token)
   @client = github_client(access_token)
end
``` 
And at the methods that currently use the `github_client` method, now will be using the instance variable `@client`, in the example the `get_milestone` will have this `miles = @client.list_milestones(repository)`.

Are my assumptions correct at that matter?

Reply:
> Are my assumptions correct at that matter?

Yes, totally 👍 

Additionally, you could also inline the implementation of the current `def github_client(access_token)` method directly in `def initialize(access_token)`, and thus get rid of the `def github_client(access_token)` method. 🧹🧹 🧹

Reply:
PS: I'd suggest you to use named parameters (on `initialize` but also on other methods you might introduce or adjust), to make the call site more explicit.

There's quite a bit of legacy in the `release-toolkit` with many of its code being written back when Ruby was in earlier versions, or when weren't so used in writing idiomatic Ruby ourselves, so you might find a couple of choices in the existing code that might not have been the best choices at the time, or which are still respecting the latest conventions, so don't hesitate to improve it to use better and nicer Ruby conventions when it makes sense, even if other code in the toolkit still doesn't.

Reply:
hahaha Yeah, for sure we improve 😆,  

Good that you comment on this as I was about to ask if I should also change the params like the `prerelease` to `pre_release` to make the methods follow the snake_case convention. This comment already answers that 😄

Reply:
> Good that you comment on this as I was about to ask if I should also change the params like the prerelease to pre_release to make the methods follow the snake_case convention. This comment already answers that 😄

Well for that one it's a bit more controversial / debatable though, because `prerelease` is also the current name of the option in `CreateReleaseAction`, and changing it there would impact the API of the action and force us to update the parameters at call site in the client apps' `Fastfile`.
You could still consider updating `prerelease` to `pre_release` in the private methods and the helper… but not on the public-facing API (the action), to not make it a breaking change, but then that would mean having an inconsistency in the parameter naming between the action and the helper (which might not be a big deal, but 🤷 ), so… up to you to decide if it's worth it 🙂